ambry-commons/src/main/java/com.github.ambry.commons/ByteBufferAsyncWritableChannel.java

```diff
@@ -31,10 +31,46 @@
  * retrieved and resolved by an external thread.
  */
 public class ByteBufferAsyncWritableChannel implements AsyncWritableChannel {
+
+  /**
+   * List of events of interest to the consumer of the content in this channel.
+   */
+  public enum EventType {
+    ChunkArrival,
+    Close,
+  }
+
+  /**
+   * Callback that can be used to listen to events that happen inside this channel.
+   */
+  public interface ChannelEventListener {
```

Comment:
Yes - the interface and enum are `ByteBufferAsyncWritableChannel` specific, as the event types may not be generic across all `AsyncWritableChannel` implementations. An implementation could have other events and those could be of interest to the creator of of that channel (who registers the listener).